### ISSUE TITLE
Dismiss on tap with block and show duration sometimes calls block twice. 

### ISSUE THREAD

**Issue by hamitron** [bryx-inc/BRYXBanner#75]
So I'm using the banner to notify users about drive space remaining before they can record video, so it will interrupt the display of a UIImagePickerController and say that memory is low.  

If you let the banner dismiss itself after its specified time, it will run the block correctly.
If you tap on it, it will dismiss the banner, and display the camera correctly, however once you dismiss the camera it brings the camera up again.

Here's the banner building:
```swift
func buildBanner(forType type:ECWarningType, withContinue continueText:Bool) -> Banner? {
	var banner: Banner?
	let (title, baseMessage) = getTitleAndSubtitleTextForWarningType(type)
	let detailMessage = continueText ? baseMessage + " Tap to continue." : baseMessage
	switch type {
	case .low:
		banner = Banner(title: title, subtitle: detailMessage, image: nil, backgroundColor: ECWarningColor.low, didTapBlock: nil)
	case .dangerous:
		banner = Banner(title: title, subtitle: detailMessage, image: nil, backgroundColor: ECWarningColor.dangerous, didTapBlock: nil)
	case .critical:
		banner = Banner(title: title, subtitle: detailMessage , image: nil, backgroundColor: ECWarningColor.critical, didTapBlock: nil)
	case .disabled:
		banner = Banner(title: title, subtitle: detailMessage, image: nil, backgroundColor: ECWarningColor.disabled, didTapBlock: nil)
	default:
		break
	}
	return banner
}

if let banner = instance.buildBanner(forType: appropriateWarning, withContinue: true) {
	banner.position = .top
	banner.dismissesOnTap = true
	if (appropriateWarning != .disabled) {
		banner.didDismissBlock = completion
	}
	banner.show(duration: 4)
}
```

The completion block passed in:

```swift
// Test to see if memory warning is necessary.
ECMemoryWarning.testForWarning { [unowned self] Void in
	if UIImagePickerController.isCameraDeviceAvailable(.rear) {
		let imagePickerController = LandscapePickerController()
		self.prepareImagePickerController(imagePickerController)
		self.parentViewController?.present(imagePickerController, animated: true, completion: nil)
	} else {
		self.createFakeVideoInDevelopment()
	}
}
```

**Comment by billzeh** [comment]
So, it's calling the didDismissBlock twice? I assume once because of the tap or swipe to dismiss and again when the duration times out? The PR looks perfectly reasonable to me at first glance.

**Comment by nmanoogian** [comment]
Yes, it looks like we're calling `dismiss` from `didTap` because `dismissesOnTap` is true. Then it's called again from the `DispatchQueue.main.asyncAfter` (even though it's already dismissed).

**Comment by relisiuol** [comment]
Fixed: #115
Please close it.
